image-loader/app/model/QuarantineUploader.scala

```diff
@@ -0,0 +1,120 @@
+package model
+
+
+import com.gu.mediaservice.lib.argo.ArgoHelpers
+import com.gu.mediaservice.lib.auth.Authentication
+import com.gu.mediaservice.lib.auth.Authentication.Principal
+import com.gu.mediaservice.lib.aws.{S3Object, UpdateMessage}
+import com.gu.mediaservice.lib.imaging.ImageOperations
+import com.gu.mediaservice.lib.logging.{LogMarker, Stopwatch, addLogMarkers}
+import com.gu.mediaservice.lib.logging.MarkerMap
+import com.gu.mediaservice.lib.resource.FutureResources._
+import com.gu.mediaservice.model._
+import lib.ImageLoaderConfig
+import lib.storage.QuarantineStore
+import net.logstash.logback.marker.LogstashMarker
+import org.joda.time.DateTime
+import play.api.Logger
+import play.api.libs.json.{JsObject, Json}
+import com.gu.mediaservice.lib.formatting._
+import model.upload.UploadRequest
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import scala.concurrent.{ExecutionContext, Future}
+
+case class QuarantineUploadOpsCfg(quarantineFileBucket: String)
+
+case class QuarantineImageUploadOpsDependencies(
+                                       config: QuarantineUploadOpsCfg,
+                                       imageOps: ImageOperations,
+                                       storeFile: UploadRequest => Future[S3Object])
+
+object QuaratineUploader{   
+
+  def toQuarantineUploadOpsCfg(config: ImageLoaderConfig): QuarantineUploadOpsCfg = {
+    QuarantineUploadOpsCfg(config.quarantineBucket)
+  }
+
+  def fromQuarantineUploadRequestShared(uploadRequest: UploadRequest, deps: QuarantineImageUploadOpsDependencies)
+                             (implicit ec: ExecutionContext, logMarker: LogMarker): Future[JsObject] = {
+
+    Logger.info("Starting image ops")
+    
+      uploadAndStoreQuarantineImage(
+        deps.storeFile,
+        uploadRequest
+        )
+  }
+
+  private def uploadAndStoreQuarantineImage(
+                                  storeFile: UploadRequest => Future[S3Object],
+                                  uploadRequest: UploadRequest
+                              )
+                                 (implicit ec: ExecutionContext, logMarker: LogMarker): Future[JsObject] = {
+    Logger.info("storing quarantine file")
+      val sourceStoreFuture = storeFile(uploadRequest)
+       val s3Source = for{
+                        s3Source <- sourceStoreFuture
+                      }yield (s3Source)
+      s3Source.map(result => Json.obj("status" -> "uploaded to quarantine bucket")) recover {case e => Json.obj("error" -> e.getMessage)}
```

Comment:
I don't think this `map` adds anything here and as discussed (at the call site) the `recover` discards the `Failure` state. Furthermore, the `Success` value that is returned is discarded anyway. Perhaps this function best returns a `Future[Unit]`?